code/datums/circuits.dm

```diff
@@ -0,0 +1,122 @@
+#define ALPHA 1
+#define BETA 2
+#define GAMMA 4
+#define DELTA 8
+#define ETA 16
+#define THETA 32
+#define IOTA 64
+#define CHOOSE_FUSES 5 //Change this to affect how many fuses are needed to make a board.
+
+/*
+* A NOTE ON EDITING: We use 7 choose 5 to give 21 combinations. Be careful about expanding this. 9 choose 5 = 126 which would be insufferable to research through.
+* You can increase the number of fuses needed by editing CHOOSE_FUSES. Increase the number of fuse options by adding fuse_point_names in the datum and newhash proc
+* You can easily change the possible boards by adding, changing, or subtracting from the possible boards list, but if you go over the combination max (default 21), it will cause an infinite loop.
+*/
+/datum/circuits
+	var/atom/holder = null //Which specific board are we pointing at?
+	var/list/fuse_point_names = list("Alpha" = ALPHA, "Beta" = BETA, "Gamma" = GAMMA, "Delta" = DELTA, "Eta" = ETA, "Theta" = THETA, "Iota" = IOTA)
+	var/list/possible_boards = list(/obj/item/weapon/circuitboard/autolathe,/obj/item/weapon/circuitboard/seed_extractor,/obj/item/weapon/circuitboard/conveyor,/obj/item/weapon/circuitboard/air_alarm,/obj/item/weapon/circuitboard/fire_alarm,/obj/item/weapon/circuitboard/airlock,/obj/item/weapon/circuitboard/power_control,/obj/item/weapon/circuitboard/vendomat,/obj/item/weapon/circuitboard/microwave)
+	var/global/list/assigned_boards = list()
+	//Each bitflag points to a board!
+	var/localbit = 0 //What are WE programmed to? Always start as 0
+
+	var/table_options = " align='center'"
+	var/row_options1 = " width='80px'"
+	var/row_options2 = " width='80px'"
+	var/window_x = 240
+	var/window_y = 300
+
+/datum/circuits/New(var/atom/homeboard)
+	..()
+	holder = homeboard
+	if(!(src.assigned_boards.len))
+		generate_schema()
+
+/datum/circuits/proc/generate_schema()

```

Comment:
Wait, does this mean all circuit blanks require different fuse setups?
If so, that's going to be painful as all hell to use.